contracts/finance/Vesting.sol

```diff
@@ -0,0 +1,410 @@
+     * @notice Retrieves vesting data by ID.
+     * @param vestingId_ The ID of the vesting contract to retrieve.
+     * @return VestingData struct.
+     */
+    function getVesting(uint256 vestingId_) public view virtual returns (VestingData memory) {
+        return _vestings[vestingId_];
+    }
+
+    /**
+     * @notice Retrieves all vesting data for a beneficiary.
+     * @param beneficiary_ The address of the beneficiary.
+     * @return An array of VestingData struct.
+     */
+    function getVestings(address beneficiary_) public view virtual returns (VestingData[] memory) {
+        uint256[] memory ids_ = _beneficiaryIds[beneficiary_].values();
+        VestingData[] memory beneficiaryVestings_ = new VestingData[](ids_.length);
+
+        for (uint256 i = 0; i < ids_.length; i++) {
+            beneficiaryVestings_[i] = _vestings[ids_[i]];
+        }
+
+        return beneficiaryVestings_;
+    }
+
+    /**
+     * @notice Retrieves all vesting IDs for a beneficiary.
+     * @param beneficiary_ The address of the beneficiary.
+     * @return An array of uint256 representing all vesting IDs for the beneficiary.
+     */
+    function getVestingIds(address beneficiary_) public view virtual returns (uint256[] memory) {
+        return _beneficiaryIds[beneficiary_].values();
+    }
+
+    /**
+     * @notice Retrieves the vested amount for a vesting ID.
+     * @param vestingId_ The ID of the vesting contract.
+     * @return The amount of tokens vested.
+     */
+    function getVestedAmount(uint256 vestingId_) public view virtual returns (uint256) {
+        VestingData storage _vesting = _vestings[vestingId_];
+        Schedule storage _schedule = _schedules[_vesting.scheduleId];
+
+        return _getVestedAmount(_vesting, _schedule, block.timestamp);
+    }
+
+    /**
+     * @notice Retrieves the withdrawable amount for a vesting ID.
+     * @param vestingId_ The ID of the vesting contract.
+     * @return The amount of tokens available to withdraw.
+     */
+    function getWithdrawableAmount(uint256 vestingId_) public view virtual returns (uint256) {
+        VestingData storage _vesting = _vestings[vestingId_];
+        Schedule storage _schedule = _schedules[_vesting.scheduleId];
+
+        return _getWithdrawableAmount(_vesting, _schedule, block.timestamp);
+    }
+
+    /**
+     * @notice Withdraws funds from a vesting contract.
+     * @param vestingId_ The ID of the vesting contract.
+     */
+    function withdrawFromVesting(uint256 vestingId_) public virtual {
+        VestingData storage _vesting = _vestings[vestingId_];
+
+        require(
+            msg.sender == _vesting.beneficiary,
+            "VestingWallet: only beneficiary can withdraw from his vesting"
+        );
+        require(_vesting.vestingAmount > 0, "VestingWallet: vesting amount is zero");
+        require(
+            _vesting.paidAmount < _vesting.vestingAmount,
+            "VestingWallet: nothing to withdraw"
+        );
+        require(
```

Comment:
Same here

Reply:
Here we just check if paidAmount less then vesting amount to see if vesting is fully paid or not